### PR TITLE
Add `--plugin-dir` flag to docs

### PR DESCRIPTION
**Pull request by laviniat1996**
Updated the docs to include the `--plugin-dir` flag for AvalancheGo, as per https://github.com/ava-labs/avalanchego/releases/tag/v1.9.6.

### CODE REVIEW

File: docs/nodes/maintain/avalanchego-config-flags.md [hunk 1]
```diff
@@ -1308,6 +1308,10 @@ GiB).
 
 If true, runs the node as a [plugin.](https://github.com/hashicorp/go-plugin) Defaults to `false`.
 
+#### `--plugin-dir` (string)
+
+Sets the directory for [plugins](https://github.com/hashicorp/go-plugin). The default value is `[DATADIR]/plugins`.
```
Comment (ceyonur) [inline docs/nodes/maintain/avalanchego-config-flags.md:1313]: AvalancheGo does not use `hashicorp/go-plugin` for plugins anymore. Can you rename `### Plugin Mode` to `### Plugins` and then reword `Sets the directory for [plugins](https://github.com/hashicorp/go-plugin)` to something like `Sets the directory for [VM plugins](../../subnets/introduction-to-vm.md)`.

File: docs/nodes/maintain/avalanchego-config-flags.md [hunk 2]
```diff
@@ -1308,6 +1308,10 @@ GiB).
 
 If true, runs the node as a [plugin.](https://github.com/hashicorp/go-plugin) Defaults to `false`.
```
Comment (ceyonur) [inline docs/nodes/maintain/avalanchego-config-flags.md:1309]: Can you remove this `--plugin-mode-enabled` flag as it's not used anymore?

File: docs/nodes/maintain/avalanchego-config-flags.md [hunk 3]
```diff
@@ -1308,6 +1308,10 @@ GiB).
...
 
+#### `--plugin-dir` (string)
```
Comment (ceyonur) [inline docs/nodes/maintain/avalanchego-config-flags.md:1311]: I believe there are some places refer `--build-dir` flag and `#build-directory` section in the repo. Can you update them with this flag?

**Review by ceyonur — COMMENTED** [review]
Thank you for working on this. I left some comments.

Great catch 🙌

File: docs/nodes/maintain/avalanchego-config-flags.md (hunk 1 above)
Comment (laviniat1996) [reply docs/nodes/maintain/avalanchego-config-flags.md]: fixed  bd4506785f53e2afec17e4bb4f63ee3ef87b66a9

File: docs/nodes/maintain/avalanchego-config-flags.md (hunk 2 above)
Comment (laviniat1996) [reply docs/nodes/maintain/avalanchego-config-flags.md]: fixed 4abab397e720b274ee80650ca44f211076a68a67

File: docs/nodes/maintain/avalanchego-config-flags.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (laviniat1996) [reply docs/nodes/maintain/avalanchego-config-flags.md]: fixed bd4506785f53e2afec17e4bb4f63ee3ef87b66a9

File: docs/nodes/maintain/avalanchego-config-flags.md (hunk 3 above)
Comment (laviniat1996) [reply docs/nodes/maintain/avalanchego-config-flags.md:1307]: fixed a875a304a2d11e4596b23bfd31856db8772e959a

File: docs/subnets/create-a-vm-timestampvm.md [hunk 4]
```diff
@@ -1251,8 +1251,7 @@ JSON file at `~/.avalanchego/configs/vms/aliases.json` with:
 
 ### Installing a VM
 
-AvalancheGo searches for and registers plugins under the `plugins` directory of the
-[build directory](../nodes/maintain/avalanchego-config-flags#build-directory).
+AvalancheGo searches for and registers plugins under the `plugins` [directory](../nodes/maintain/avalanchego-config-flags#plugins).
```
Comment (ceyonur) [inline docs/subnets/create-a-vm-timestampvm.md:1254]: Just a nit
```suggestion
AvalancheGo searches for and registers plugins under the `plugins` [directory](../nodes/maintain/avalanchego-config-flags#--plugin-dir-string).
```

**Review by ceyonur — COMMENTED** [review]
just a small nit then lgtm

File: docs/subnets/create-a-simple-rust-vm.md [hunk 5]
```diff
@@ -1237,8 +1237,7 @@ async fn main() -> io::Result<()> {
 
 ### Installing a VM
 
-AvalancheGo searches for and registers VM plugins under the `plugins` directory of the [build
-directory](../nodes/maintain/avalanchego-config-flags#build-directory).
+AvalancheGo searches for and registers VM plugins under the `plugins` [directory](../nodes/maintain/avalanchego-config-flags#--plugin-dir-string).
```
Comment (yulin-dong) [inline docs/subnets/create-a-simple-rust-vm.md:1240]:
```suggestion
AvalancheGo searches for and registers VM plugins under the `plugins` [directory](../nodes/maintain/avalanchego-config-flags.md#--plugin-dir-string).
```

File: docs/nodes/maintain/avalanchego-config-flags.md (hunk 3 above)
Comment (ceyonur) [reply docs/nodes/maintain/avalanchego-config-flags.md:1307]: It looks good. Just a small nit left. Thanks for noticing this missing flag!

File: docs/subnets/create-a-vm-timestampvm.md (hunk 4 above)
Comment (yulin-dong) [reply docs/subnets/create-a-vm-timestampvm.md:1254]: add `.md` to the file name as in `../nodes/maintain/avalanchego-config-flags.md#--plugin-dir-string`

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**